src/providers/mbed_provider/asym_encryption.rs

```diff
@@ -0,0 +1,101 @@
+        let key_name = op.key_name.clone();
+
+        let key_triple = KeyTriple::new(app_name, ProviderID::MbedCrypto, key_name);
+        let store_handle = self.key_info_store.read().expect("Key store lock poisoned");
+        let key_id = key_management::get_key_id(&key_triple, &*store_handle)?;
+        let _guard = self
+            .key_handle_mutex
+            .lock()
+            .expect("Grabbing key handle mutex failed");
+        let id = key::Id::from_persistent_key_id(key_id);
+        let key_attributes = key::Attributes::from_key_id(id)?;
+
+        op.validate(key_attributes)?;
+        let salt_buff = match &op.salt {
+            Some(salt) => Some(salt.as_slice()),
+            None => None,
+        };
+        let alg = op.alg;
+        let buffer_size = key_attributes.asymmetric_encrypt_output_size(alg)?;
+        let mut ciphertext = vec![0u8; buffer_size];
+
+        match asym_encryption::encrypt(
+            id,
+            alg,
+            &op.plaintext.expose_secret(),
+            salt_buff,
+            &mut ciphertext,
+        ) {
+            Ok(output_size) => {
+                ciphertext.resize(output_size, 0);
+                Ok(psa_asymmetric_encrypt::Result { ciphertext })
+            }
+            Err(error) => {
+                let error = ResponseStatus::from(error);
+                format_error!("Encrypt status: {}", error);
+                Err(error)
+            }
+        }
+    }
+
+    pub(super) fn psa_asymmetric_decrypt_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_asymmetric_decrypt::Operation,
+    ) -> Result<psa_asymmetric_decrypt::Result> {
+        info!("Mbed Provider - Asym Decrypt");
+
+        let key_triple = KeyTriple::new(app_name, ProviderID::MbedCrypto, op.key_name.clone());
+        let store_handle = self.key_info_store.read().expect("Key store lock poisoned");
+        let key_id = key_management::get_key_id(&key_triple, &*store_handle)?;
+
+        let _guard = self
+            .key_handle_mutex
+            .lock()
+            .expect("Grabbing key handle mutex failed");
+
+        let id = key::Id::from_persistent_key_id(key_id);
+        let key_attributes = key::Attributes::from_key_id(id)?;
+
+        op.validate(key_attributes)?;
+        let salt_buff = match &op.salt {
+            Some(salt) => Some(salt.as_slice()),
+            None => None,
+        };
+        let buffer_size = key_attributes.asymmetric_decrypt_output_size(op.alg)?;
+        let mut plaintext = vec![0u8; buffer_size];
+
+        match asym_encryption::decrypt(id, op.alg, &op.ciphertext, salt_buff, &mut plaintext) {
+            Ok(output_size) => {
+                plaintext.resize(output_size, 0);
+                Ok(psa_asymmetric_decrypt::Result { plaintext })
+            }
+            Err(error) => {
+                // When dropped, will zero buffer incase anything was written before error
+                let _ = Secret::new(plaintext);
```

Comment:
Easier/quicker to just `plaintext.zeroize()` here (though you need to import the `Zeroize` trait).

Reply:
huh, I didn't know you could do that 🤯! How does it work? `plaintext` is a regular `Vec<u8>' and I can't see anything obvious in the documentation that states `Zeroizable` is implemented for `Vec<u8>`. Is it `u8` implements `DefaultIsZeros`, so implements `Zeroizable` in the blanket implementation and `Vec<u8>` 'inherits' (derefs?) that?

Reply:
[`Zeroize` is implemented on `Vec`s](https://docs.rs/zeroize/1.1.0/zeroize/trait.Zeroize.html#impl-Zeroize-for-Vec%3CZ%3E) - it's actually what you were using in the other PR when doing `plaintext.zeroize()` for the protobuf types - those types don't use `Zeroizing` or anything from the crate

Reply:
Oh yeah... 🤦‍♂️